src/AutofacSerilogIntegration/ContextualLoggingModule.cs

```diff
@@ -48,8 +50,17 @@ protected override void Load(ContainerBuilder builder)
 
                 return logger;
             })
-            .As<ILogger>()
-            .ExternallyOwned();
+            .As<ILogger>();
+
+            if (_dispose)
+            {
+                registration.SingleInstance()
```

Comment:
Thinking about it some more, the last scenario could be a valid one even if the RegisterLogger is passed an ILogger, the hosting application might prefer NOT having to further care about disposing the root logger and prefer leaving it up to the container